### PR TITLE
New PMCD API for multiple ProjectVersion

### PR DESCRIPTION
**Pull request by abhishoya-gs**
New API for getting PMCD for multiple GAVs

### CODE REVIEW

File: legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/services/pure/model/context/PureModelContextServiceImpl.java [hunk 1]
```diff
@@ -71,6 +72,14 @@ public PureModelContextData getPureModelContextData(String groupId, String artif
         });
     }
 
+    @Override
+    public PureModelContextData getDependenciesPureModelContextData(List<ProjectVersion> projectDependencies, String clientVersion, boolean transitive, boolean includeOrigin)
```
Comment (gs-gunjan) [inline legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/services/pure/model/context/PureModelContextServiceImpl.java:76]: can we rename this to getPureModelContextData(List<ProjectVersion>..) it becomes self explanatory

File: legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/services/pure/model/context/PureModelContextServiceImpl.java [hunk 2]
```diff
@@ -81,11 +90,16 @@ protected String resolveAndValidateClientVersion(String clientVersion)
     }
 
     protected PureModelContextData getPureModelContextData(Stream<Entity> entities, String groupId, String artifactId, String versionId, String clientVersion)
+    {
+        return getPureModelContextData(entities, buildAlloySDLC(groupId, artifactId, versionId), clientVersion);
```
Comment (gs-gunjan) [inline legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/services/pure/model/context/PureModelContextServiceImpl.java:94]: rename this to buildPureModelContextData

File: legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/server/resources/pure/model/context/PureModelContextResource.java [hunk 3]
```diff
@@ -66,5 +71,25 @@ public Response getPureModelContextData(@PathParam("groupId") String groupId,
...
+    @POST
+    @Path("projects/dependencies/pureModelContextData")
+    @ApiOperation(GET_VERSIONS_DEPENDENCY_ENTITIES_AS_PMCD)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getPureModelContextData(@ApiParam("projectDependencies") List<ProjectVersion> projectDependencies,
+                                            @QueryParam("clientVersion") String clientVersion,
+                                            @QueryParam("transitive") @DefaultValue("true")
+                                            @ApiParam("Whether to return transitive dependencies") boolean transitive,
+                                            @QueryParam("includeOrigin") @DefaultValue("true")
+                                                @ApiParam("Whether to return start of dependency tree") boolean includeOrigin,
```
Comment (gs-gunjan) [inline legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/server/resources/pure/model/context/PureModelContextResource.java:83]: do we need includeOrigin at all, do we have a use case when its false?

File: legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/server/resources/pure/model/context/PureModelContextResource.java [hunk 4]
```diff
@@ -66,5 +71,25 @@ public Response getPureModelContextData(@PathParam("groupId") String groupId,
...
+                                            @Context Request request)
+    {
+        return handle(GET_VERSIONS_DEPENDENCY_ENTITIES_AS_PMCD, () -> service.getDependenciesPureModelContextData(projectDependencies, clientVersion, transitive, includeOrigin), request, () ->
+        {
```
Comment (gs-gunjan) [inline legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/server/resources/pure/model/context/PureModelContextResource.java:87]: will the etag still work if the project dependencies are the same but in different order?

File: legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/services/pure/model/context/PureModelContextServiceImpl.java (hunk 1 above)
Comment (abhishoya-gs) [reply legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/services/pure/model/context/PureModelContextServiceImpl.java]: done

File: legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/services/pure/model/context/PureModelContextServiceImpl.java (hunk 2 above)
Comment (abhishoya-gs) [reply legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/services/pure/model/context/PureModelContextServiceImpl.java]: changed to build

File: legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/server/resources/pure/model/context/PureModelContextResource.java (hunk 3 above)
Comment (abhishoya-gs) [reply legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/server/resources/pure/model/context/PureModelContextResource.java]: not really, removed

File: legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/server/resources/pure/model/context/PureModelContextResource.java (hunk 4 above)
Comment (abhishoya-gs) [reply legend-depot-pure-model-context/src/main/java/org/finos/legend/depot/server/resources/pure/model/context/PureModelContextResource.java]: no, removed for now

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**